src/snippet/mod.rs

```diff
@@ -280,7 +282,17 @@ impl SnippetGenerator {
     pub fn snippet_from_doc(&self, doc: &Document) -> Snippet {
         let text: String = doc
             .get_all(self.field)
-            .flat_map(Value::as_text)
+            .flat_map(|v| match v {
+                Value::Str(text) => Some(vec![text.as_str()]),
+                Value::JsonObject(json_object) => Some(
+                    json_object
+                        .values()
+                        .flat_map(|v| v.as_str())
```

Comment:
This only works over one level of nesting.

Reply:
Oh that's true...!  Let me rework